libpkg/pkg_add.c

```diff
@@ -351,9 +352,16 @@ do_extract_dir(struct pkg* pkg, struct archive *a __unused, struct archive_entry
 			return (EPKG_FATAL);
 		}
 	}
-	if (st.st_uid == d->uid && st.st_gid == d->gid &&
-	    (st.st_mode & S_IFMT) == (d->perm & S_IFMT)) {
-		d->noattrs = true;
+
+	asprintf(&metalogentry,
```

Comment:
this is metalog specific thing that is "copy/pasted" all over the place, this should be done in metalog_add

Please you are leaking memory here given that asprintf is allocating memory that you never free